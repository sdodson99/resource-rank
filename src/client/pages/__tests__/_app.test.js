import React from 'react';
import App from '../_app.page';
import { createRenderer } from 'react-test-renderer/shallow';
import { useRouter } from 'next/router';

jest.mock('next/router');

describe('<App />', () => {
  let Component;

  beforeEach(() => {
    Component = function MockComponent() {
      return <div>Test component</div>;
    };
  });

  afterEach(() => {
    useRouter.mockReset();
  });

  it('should render correctly', () => {
    useRouter.mockReturnValue({ asPath: '/path' });

    const page = createRenderer().render(<App Component={Component} />);

    expect(page).toMatchSnapshot();
  });
});
