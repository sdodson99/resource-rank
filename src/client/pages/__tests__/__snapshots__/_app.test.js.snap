// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<Constate
  firebaseConfig={
    Object {
      "apiKey": "AIzaSyChFnYmkhARBy0Hwtehlx-81rSC7PZZWT8",
      "appId": "1:489304913015:web:71f38a965ccde83db54272",
      "authDomain": "resource-rank.firebaseapp.com",
      "databaseURL": "https://resource-rank-default-rtdb.firebaseio.com",
      "measurementId": "G-D18DVJ4168",
      "messagingSenderId": "489304913015",
      "projectId": "resource-rank",
      "storageBucket": "resource-rank.appspot.com",
    }
  }
  useAuthenticationEmulator={false}
>
  <Constate>
    <Constate>
      <DefaultSeo
        description="Find the best resources to simplify your learning journey."
        openGraph={
          Object {
            "description": "Find the best resources to simplify your learning journey.",
            "images": Array [
              Object {
                "alt": "Resource Rank Logo",
                "height": 630,
                "url": "http://undefined/img/og-logo.jpg",
                "width": 1200,
              },
            ],
            "title": "Simplify Learning - Resource Rank",
            "type": "website",
            "url": "https://undefined/path",
          }
        }
        title="Simplify Learning"
        titleTemplate="%s - Resource Rank"
      />
      <MockComponent />
    </Constate>
  </Constate>
</Constate>
`;
